### PR TITLE
[Fix] isLabelSyncedWithName should be nullable

### PR DESCRIPTION
**Pull request by ijreilly**
isLabelSyncedWithName should be nullable for fieldMetadata, as it is for objectMetadata.

+ Adding missing validation on label and name sync in fieldMetadataService for creation and update
+ adding metadata tests

### CODE REVIEW

File: packages/twenty-server/src/engine/metadata-modules/field-metadata/dtos/field-metadata.dto.ts [hunk 1]
```diff
@@ -147,7 +147,7 @@ export class FieldMetadataDTO<
 
   @IsBoolean()
   @IsOptional()
-  @Field()
+  @Field({ nullable: true })
```
Comment (ijreilly) [inline packages/twenty-server/src/engine/metadata-modules/field-metadata/dtos/field-metadata.dto.ts:150]: this is nullable in the input but non-null in db, as it is for objectMetadata. I really need to double check but I think we want to allow null values in the input that we backfill in the service

File: packages/twenty-server/test/integration/metadata/suites/field-metadata/update-one-field-metadata.integration-spec.ts [hunk 2]
```diff
@@ -0,0 +1,107 @@
...
+            name
+            label
+            isLabelSyncedWithName
+        `,
+      });
+
+      const response = await makeMetadataAPIRequest(graphqlOperation);
+
+      // Assert
+      console.log(
```
Comment (Weiko) [inline packages/twenty-server/test/integration/metadata/suites/field-metadata/update-one-field-metadata.integration-spec.ts:50]: are those logs intended?

**Review by Weiko — APPROVED** [review]
LGTM!

File: packages/twenty-server/test/integration/metadata/suites/field-metadata/update-one-field-metadata.integration-spec.ts (hunk 2 above)
Comment (ijreilly) [reply packages/twenty-server/test/integration/metadata/suites/field-metadata/update-one-field-metadata.integration-spec.ts:50]: oopsie !

**Pull request merged**